### PR TITLE
Add support for ContentDialog

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): Fixes https://github.com/unoplatform/uno/issues/124

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

Adds support for `ContentDialog`.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialogPopupPanel.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		private readonly ContentDialog _dialog;
+
+		public ContentDialogPopupPanel(ContentDialog dialog) : base(dialog._popup)
+		{
+			_dialog = dialog;
+		}
+
+		protected override Size ArrangeOverride(Size finalSize)
+		{
+			Console.WriteLine($"ArrangeOverride ContentDialogPopupPanel {finalSize}");
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialogPopupPanel.cs]: change to logging

File: src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialogPopupPanel.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+			foreach (var child in Children)
+			{
+				if (!(child is UIElement elem))
+				{
+					continue;
+				}
+
+				var desiredSize = elem.DesiredSize;
+				var rect = CalculateDialogPlacement(desiredSize);
+				Console.WriteLine($"Arranging ContentDialogPopupPanel {desiredSize} to {rect}");
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialogPopupPanel.cs]: logging

File: src/Uno.UI/Uno.UI.csproj [hunk 3]
```diff
@@ -357,91 +360,10 @@
...
-	    <Generator>MSBuild:Compile</Generator>
-	  </None>
-	  <None Update="UI\Xaml\Style\Generic\Generic.Native.xaml">
-	    <Generator>MSBuild:Compile</Generator>
-	  </None>
-	  <None Update="UI\Xaml\Style\Generic\Generic.xaml">
-	    <Generator>MSBuild:Compile</Generator>
-	  </None>
-	  <None Update="UI\Xaml\Style\Generic\SemanticStylesResources.xaml">
+	  <None Update="C:\s\nv.github\uno\src\Uno.UI\UI\Xaml\Controls\ContentDialog\ContentDialog.xaml">
```
Comment (davidjohnoliver) [inline src/Uno.UI/Uno.UI.csproj]: sanitize the changes here to the `csproj`

File: src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.cs [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+		}
+
+		protected override void OnApplyTemplate()
+		{
+			base.OnApplyTemplate();
+
+			UpdateButtonsVisualStates();
+		}
+
+		public IAsyncOperation<ContentDialogResult> ShowAsync()
```
[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.cs]: Good catch, here's what happens: 
image
 :)

File: doc/articles/uno-development/working-with-the-samples-apps.md [hunk 5]
```diff
@@ -39,6 +39,9 @@ At this time, only WebAssembly and Android are used to run UI Tests, iOS is comi
 - Open the [`Constants.cs](src/SamplesApp/SamplesApp.UITests/Constants.cs) file and change the `CurrentPlatform` field to the platform you want to test.
 - Select a test in the `SamplesApp.UITests` project and run a specific test.
 
+## Troubleshooting tests running during the CI
+The build output contains the
```
Comment (MatFillion) [inline doc/articles/uno-development/working-with-the-samples-apps.md]: This PR is merged already, but @jeromelaban there is unfinished sentence here.

Comment (jeromelaban) [reply doc/articles/uno-development/working-with-the-samples-apps.md]: I fixed it afterwards :)

**Pull request merged**